scripts/upgrade

```diff
@@ -102,146 +80,133 @@ ynh_abort_if_errors
 #=================================================
-ynh_script_progression --message="Stopping a systemd service..." --time --weight=1
+ynh_script_progression --message="Stopping a systemd service..." --weight=1
 
-ynh_systemd_action --service_name=$app --action="stop" --log_path="$log_filename"
-
-#=================================================
-# DOWNLOAD, CHECK AND UNPACK SOURCE
-#=================================================
-
-if [ "$upgrade_type" == "UPGRADE_APP" ]
-then
-	ynh_script_progression --message="Upgrading source files..." --time --weight=1
-
-	# Download, check integrity, uncompress and patch the source from app.src
-	ynh_setup_source --dest_dir="$final_path/src"
-fi
+ynh_systemd_action --service_name=$app --action="stop"
 
 #=================================================
 # UPGRADE DEPENDENCIES
 #=================================================
-ynh_script_progression --message="Upgrading dependencies..." --time --weight=1
+ynh_script_progression --message="Upgrading dependencies..." --weight=1
 
-ynh_install_app_dependencies $pkg_dependencies
+ynh_exec_warn_less ynh_install_app_dependencies $pkg_dependencies
 
+# libolm-dev avoided by pre-compiled python-olm
+ynh_install_extra_app_dependencies --repo="http://http.debian.net/debian buster-backports main" --package="libolm-dev" --key="https://keyserver.ubuntu.com/pks/lookup?search=0x0E98404D386FA1D9&op=get"
 ynh_install_extra_app_dependencies --repo="https://updates.signald.org unstable main" --package="$extra_dependencies" --key="https://updates.signald.org/apt-signing-key.asc"
 
 #=================================================
 # CREATE DEDICATED USER
 #=================================================
-ynh_script_progression --message="Making sure dedicated system user exists..." --time --weight=1
+ynh_script_progression --message="Making sure dedicated system user exists..." --weight=1
 
 # Create a dedicated user (if not existing)
-ynh_system_user_create --username=$mautrix_bridge_user
+ynh_system_user_create --username=$app
 
 #=================================================
-# SPECIFIC UPGRADE
-#=================================================
-# ...
+# DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
 
-#=================================================
-# SETUP SYSTEMD
-#=================================================
-ynh_script_progression --message="Upgrading systemd configuration..." --time --weight=1
+if [ "$upgrade_type" == "UPGRADE_APP" ]
+then
+	ynh_script_progression --message="Upgrading source files..." --weight=1
 
-# Create a dedicated systemd config
-ynh_add_systemd_config
+	# Download, check integrity, uncompress and patch the source from app.src
+	ynh_setup_source --dest_dir="$final_path/src"
+	chmod 750 $final_path
+	chown $app:$app $final_path
+fi
+
+# Fix possible permission issues with a previous signald version, esp. with stickers
+chmod -R g+rwX /var/lib/signald/{avatars,attachments,stickers}
```

Comment:
Note that this might fail if one of the folders do not exist :thinking:

Reply:
if you do it here, do it in install script too.

Reply:
Should not be needed in the install script as `signald` was fixed to create files and folders with the right permissions since then. It's only needed for files created in previous signald versions.

In the future, this won't be needed at all, as files will transit trough the signald socket, speaking with signald devs.